projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { DebugElement } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { Cart, CartService, I18nTestingModule, Voucher } from '@spartacus/core';
+import { of } from 'rxjs';
+import { CartCouponComponent } from './cart-coupon.component';
+import createSpy = jasmine.createSpy;
+
+const coupon1: Voucher = { code: 'coupon1' };
+const coupon2: Voucher = { code: 'coupon2' };
+const cart: Cart = {
+  code: 'xxx',
+  appliedVouchers: [coupon1, coupon2],
+};
+
+describe('CartCouponComponent', () => {
+  let component: CartCouponComponent;
+  let fixture: ComponentFixture<CartCouponComponent>;
+  let mockCartService: any;
+  let form: DebugElement;
+  let submit: DebugElement;
+
+  beforeEach(async(() => {
+    mockCartService = {
+      applyVoucher: createSpy(),
+      removeVoucher: createSpy(),
+      getAddVoucherResultSuccess: createSpy().and.returnValue(of({})),
+    };
+

```

Comment:
could you please change to createSpyObj (due to you multiple times to create spy)